unidist/core/backends/mpi/core/local_object_store.py

```diff
@@ -11,49 +11,53 @@
 import unidist.core.backends.mpi.core.communication as communication
 
 
-class ObjectStore:
+class LocalObjectStore:
     """
-    Class that stores objects and provides access to these from master process.
+    Class that stores local objects and provides access to them.
 
     Notes
     -----
-    Currently, the storage is local to the current process only.
+    Currently, the storage is local to the current worker process only.
     """
 
     __instance = None
 
     def __init__(self):
-        # Add local data {DataID : Data}
+        # Add local data {DataId : Data}
         self._data_map = weakref.WeakKeyDictionary()
-        # Data owner {DataID : Rank}
+        # "strong" references to data IDs {DataId : DataId}
+        # we are using dict here to improve performance when getting an element from it,
+        # whereas other containers would require O(n) complexity
+        self._data_id_map = {}
+        # Data owner {DataId : Rank}
         self._data_owner_map = weakref.WeakKeyDictionary()
         # Data was already sent to this ranks {DataID : [ranks]}
         self._sent_data_map = defaultdict(set)
         # Data id generator
         self._data_id_counter = 0
         # Data serialized cache
-        self._serialization_cache = {}
+        self._serialization_cache = weakref.WeakKeyDictionary()
```

Comment:
Whi is this changed?

Reply:
It is a GC problem, but I found it when checked shared memory cleanup.